server/src/main/java/org/hyperskill/database/serverside/NetworkService.java

```diff
@@ -0,0 +1,96 @@
+package org.hyperskill.database.serverside;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import org.hyperskill.database.common.Request;
+import org.hyperskill.database.common.Response;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+class NetworkService implements Runnable {
```

Comment:
Why making it `Runnable` if you're not using it as runnable
Calling `nws.run();` in `main` doesn't execute it in a separate thread. You need to call `start` for that